src/parsing.jl

```diff
@@ -1,267 +1,156 @@
 # Functions to parse strings to intervals
 
 """
-    parse{T}(DecoratedInterval{T}, s::AbstractString)
+    parse(DecoratedInterval, s::AbstractString)
 
 Parse a string of the form `"[a, b]_dec"` as a `DecoratedInterval`
 with decoration `dec`.
 """
 function parse(::Type{DecoratedInterval{T}}, s::AbstractString) where T
-    m = match(r"(\[.*\])(\_.*)?", s)
-
-    if m == nothing  # matched
-
-        m = match(r"(.*\?[a-z0-9]*)(\_.*)?", s)
-
-        if m == nothing
-            throw(ArgumentError("Unable to process string $x as decorated interval"))
-        end
-
+    if '_' ∉ s
+        x = parse(Interval{T}, s)
+        return DecoratedInterval(x.lo, x.hi)
     end
 
-    if m.captures[1] == "[nai]" || m.captures[1] == "[Nai]"
-        return nai(T)
-    end
+    decorations = Dict(
+        "ill" => ill,
+        "trv" => trv,
+        "def" => def,
+        "dac" => dac,
+        "com" => com)
 
-    interval_string, decoration_string = m.captures
+    interval_string, dec = split(s, "_")
     interval = parse(Interval{T}, interval_string)
-
-    # type unstable:
-    if decoration_string != nothing
-        decoration_string = lowercase(decoration_string)
-        decoration_symbol = Symbol(decoration_string[2:end])
-        decoration = getfield(IntervalArithmetic, decoration_symbol)
-        return DecoratedInterval(interval, decoration)
-    else
-        DecoratedInterval(interval)
-    end
-
+    return DecoratedInterval(interval, decorations[dec])
```

Comment:
To be safe with respect to the case used for the decorations, which I think allows some tests of the ITF1788 suite to pass, we need
```julia
DecoratedInterval(interval, decorations[lowercase(dec)])
```